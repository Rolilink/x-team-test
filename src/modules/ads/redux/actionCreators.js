import { createAction } from 'redux-actions';
import { ADD_AD, CLEAR_ADS } from './actionTypes';

export const addAd = createAction(ADD_AD, ad => ({ ad }));
export const clearAds = createAction(CLEAR_ADS);
