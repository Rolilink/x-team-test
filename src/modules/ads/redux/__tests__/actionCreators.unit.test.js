import { addAd, clearAds } from '../actionCreators';

describe('Ads.Redux.ActionCreators.addAd', () => {
  it('should return an ADD_AD action with an ad payload', () => {
    const ad = 1;

    expect(addAd(ad)).toMatchSnapshot();
  });
});

describe('Ads.Redux.ActionCreators.clearAds', () => {
  it('should return an CLEAR_ADS action', () => {
    expect(clearAds()).toMatchSnapshot();
  });
});
