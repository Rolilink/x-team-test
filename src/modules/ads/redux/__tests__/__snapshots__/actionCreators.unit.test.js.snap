// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ads.Redux.ActionCreators.addAd should return an ADD_AD action with an ad payload 1`] = `
Object {
  "payload": Object {
    "ad": 1,
  },
  "type": "x-team/discount-ascii-warehouse/ads/ADD",
}
`;

exports[`Ads.Redux.ActionCreators.clearAds should return an CLEAR_ADS action 1`] = `
Object {
  "type": "x-team/discount-ascii-warehouse/ads/CLEAR",
}
`;
